src/zenml/stack/stack.py

```diff
@@ -509,11 +508,106 @@ def validate(self) -> None:
             remote execution) specified by the orchestrator of the stack
         - the `StackValidator` of each stack component has to validate the
             stack to make sure all the components are compatible with each other
+        - the stack must either have a properly associated artifact/metadata
+            store pair or reset the association.
+
+        Args:
+            decouple_stores: Flag to reset the previous associations between
+                an artifact store and a metadata store
+
+        Raises:
+            StackValidationError: If the artifact store and the metadata store
+                are not properly associated.
         """
+        from zenml.cli.utils import warning
```

Comment:
IMO no CLI code should be imported in non-cli code, as this function is already used when not getting called from the CLI.